functions/scraper/func.py

```diff
@@ -18,52 +19,41 @@
     format='parsed-json'
 )
 
-PHOTO_SOURCE_URL = 'https://farm{0}.staticflickr.com/{1}/{2}_{3}{4}.{5}'
-
-def get_image_url(photo_dict):
-    return PHOTO_SOURCE_URL.format(
-        photo_dict['farm'], photo_dict['server'],
-        photo_dict['id'], photo_dict['secret'],
-        '_c', 'jpg'
-    )
-
-
-def photo_to_payload(body, photo_dict):
-    return {
-        "id": photo_dict.get('id'),
-        "image_url": get_image_url(photo_dict),
-        "countrycode": body.get("countrycode"),
-        "bucket": body.get("bucket", "")
-    }
-
-
 def handler(ctx, data=None, loop=None):
-    payloads = []
+    
+    # Scrape Flickr
     if data and len(data) > 0:
         body = ujson.loads(data)
+        
         photos = flickr.photos.search(
-            text=body.get("query", "baby smile"),
-            per_page=int(body.get("num", "5")),
-            page=int(body.get("page", int(random.uniform(1, 50)))),
-            extras="original_format",
-            safe_search="1",
-            content_type="1",
+                text=body.get("query", "baby smile"),
+                per_page=int(body.get("num", "5")),
+                page=int(body.get("page", int(random.uniform(1, 50)))),
+                extras="original_format",
+                safe_search="1",
+                content_type="1",
         )
 
+        # For each photo
         for p in photos.get('photos', {'photo': {}}).get('photo', []):
-            payloads.append(photo_to_payload(body, p))
-
-    for p in payloads:
-        this_payload = {
-            "media":[p.get("image_url")], 
-            "event_id": "test_id",
-            "event_type": "test_type",
-            "ran_on": "<fn-hostname-or-dns>"
-        }
-        process_url = "http://docker.for.mac.localhost:8080/t/cloudevents/image-processor"
-        r = requests.post(process_url, data=ujson.dumps(this_payload))
-    
-    return {"result": payloads}
+            photo_url_tpl = 'https://farm{0}.staticflickr.com/{1}/{2}_{3}{4}.{5}'
+            photo_url = photo_url_tpl.format(p['farm'], p['server'],p['id'], p['secret'], '_c', 'jpg')
+
+            data = {"photo_url": photo_url}
+
+            event = (
+                v02.Event().
+                SetContentType("application/json").
+                SetData(data).
+                SetEventID("my-id").
+                SetSource("scraper").
+                SetEventType("cloudevent.flickr.image")
+            )
+            event_json = ujson.dumps(event.Properties())
```

Comment:
That’s why I pointed you to ToRequest API. Here you’re basically lucky because your event data a dict, in other case it wouldn’t work.

Here you need to create an instance of an marshaller, from that you need to call a ToRequest method that will turn your event objection into serialized one (headers and a body).